server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -3304,6 +3323,13 @@ private EngineConfig newEngineConfig(LongSupplier globalCheckpointSupplier) thro
         if (this.checkpointPublisher != null && indexSettings.isSegRepEnabled() && shardRouting.primary()) {
             internalRefreshListener.add(new CheckpointRefreshListener(this, this.checkpointPublisher));
         }
+        /**
+         * With segment replication enabled for primary relocation, recover replica shard initially as read only and
+         * promote to during relocation handoff post segment replication.
+         */
+        boolean isReadOnlyReplica = indexSettings.isSegRepEnabled()
```

Comment:
nit - does the variable name sound intuitive when the shard is actually a bootstrapping primary(`shardRouting.primary() == true`). May be let's rename it?